File: java/src/atcoder/ABC103/B.java
package atcoder.ABC103;

import java.util.Scanner;

public class B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        String t = sc.next();

        if (s.equals(t)) {
            System.out.println("Yes");
            return;
        }

        int len = s.length();
        for (int i = 0; i < len-1; i++) {
            s = s.substring(len-1, len) + s.substring(0, len-1);
            if (s.equals(t)) {
                System.out.println("Yes");
                return;
            }
        }
        System.out.println("No");
    }
}

File: kotlin/src/atcoder/ABC103/B.kt
package atcoder.ABC103

fun main(args: Array<String>) {
    var s = readLine()!!
    val t = readLine()!!

    if (s == t) {
        println("Yes")
        return
    }

    fun String.rotate() = drop(length - 1) + take(length - 1)

    for (i in 1..s.length) {
        s = s.rotate()
        if (s == t) {
            println("Yes")
            return
        }
    }

    println("No")
}